lib/usagereporter/usagereporter.go

```diff
@@ -153,94 +154,59 @@ func (r *UsageReporter[T]) runSubmit(ctx context.Context) {
 	defer r.wg.Done()
 
 	for {
+		var batch []*SubmittedEvent[T]
+		var ok bool
 		select {
 		case <-ctx.Done():
 			return
-		case batch, ok := <-r.submissionQueue:
-			if !ok {
-				return
-			}
-			t0 := time.Now()
+		case batch, ok = <-r.submissionQueue:
+		}
+		if !ok {
+			return
+		}
 
-			if failed, err := r.submit(r, batch); err != nil {
-				r.WithField("batch_size", len(batch)).Warnf("failed to submit batch of usage events: %v", err)
-				usageBatchesFailed.Inc()
+		t0 := time.Now()
 
-				var resubmit []*SubmittedEvent[T]
-				for _, e := range failed {
-					e.retriesRemaining--
+		if failed, err := r.submit(r, batch); err != nil {
+			r.WithField("batch_size", len(batch)).Warnf("failed to submit batch of usage events: %v", err)
```

Comment:
I guess? I didn't change anything in `runSubmit` other than dedenting the main body instead of leaving it in the `select`, but I can change this one to report the error as a structured field (and maybe remove the error altogether in the "dropping events due to error" line, as they're both going to be the same error).